modules/network/src/main/java/org/apache/ignite/network/AbstractMessagingService.java

```diff
@@ -15,36 +15,28 @@
  * limitations under the License.
  */
 
-package org.apache.ignite.network.scalecube;
+package org.apache.ignite.network;
 
-import org.apache.ignite.network.message.NetworkMessage;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.CopyOnWriteArrayList;
 
 /**
- * Test response.
+ * Base class for {@link MessagingService} implementations.
  */
-public class TestResponse implements NetworkMessage {
-    /** Public type for tests. */
-    public static final short TYPE = 2;
+public abstract class AbstractMessagingService implements MessagingService {
+    /** */
+    private final Collection<NetworkMessageHandler> messageHandlers = new CopyOnWriteArrayList<>();
 
-    /**
-     * Some response test value.
-     */
-    private final int responseNumber;
-
-    /** Constructor. */
-    public TestResponse(int responseNumber) {
-        this.responseNumber = responseNumber;
+    /** {@inheritDoc} */
+    @Override public void addMessageHandler(NetworkMessageHandler handler) {
```

Comment:
We need a way to unregister message handlers for dynamically started components.

Reply:
Can you elaborate on the use cases please, because I may be out of context for this suggestion?

Reply:
Some components (like a cache) can start/stop dynamically and should have possibility to deregister related listeners on stop.
I'm ok to implement this as a separate ticket.

Reply:
https://issues.apache.org/jira/browse/IGNITE-14519